### PR TITLE
Lra 364 mmss devise login for faculty

### PR DESCRIPTION
**Pull request by britaumich**
There are so many changed files because the updated version of the annotate gem edited all models and specs factories and models (about 55 files).

http://localhost:3000/faculty_login - default URL for faculty 

Also please check faculty sign-in errors and other faculty messages.

### CODE REVIEW

**Review by rsmoke — APPROVED** [review]
This is very interesting work with the login redirects.
This is not a PR killer but I believe if you add a link the to the student list that is labelled "more info" or see "student details" rather than use the email as a link to the student page it would be more logical.  When I see a link that is labelled as an email address I am expecting to send an email. 
What do you think?

File: app/admin/faculties.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
+ActiveAdmin.register Faculty do
+  menu parent: 'Logins Info', priority: 1
+
+  # See permitted parameters documentation:
+  # https://github.com/activeadmin/activeadmin/blob/master/docs/2-resource-customization.md#setting-up-strong-parameters
+  #
+  # Uncomment all parameters which should be permitted for assignment
```
[the remark this answers is not in the document]
Comment (britaumich) [reply app/admin/faculties.rb:7]: I forgot to edit active_admin file for faculty. We decided that we don't need New and Edit, so I removed these actions and filters too.

**Comment by britaumich** [comment]
> What do you think?

I agree. I edited both tables (student_list and one_student). I don't think I like the second one, but I don't know what to change there.

**Comment by britaumich** [comment]
> This is very interesting work with the login redirects.

I didn't invent them - I found them on Google.

File: app/views/faculties/student_list.html.erb [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      <% else %>
+        <tr class="relative py-2 pl-4 sm:pl-6 pr-3 text-sm border-t border-grey-700 bg-white">
+      <% end %>
+        <td class="relative py-2 pl-4 sm:pl-6 pr-3 text-sm">
+          <div class="font-medium text-gray-900">
+            <%= student.applicant_detail.full_name %>
+          </div>
+        </td>
+        <td class="relative py-2 pl-4 sm:pl-6 pr-3 text-sm">
+          <div class="font-medium text-gray-900 hover:underline">
```
Comment (rsmoke) [inline app/views/faculties/student_list.html.erb:49]: Since you have the "more info" now, maybe drop the hover:underline here

**Review by rsmoke — APPROVED** [review]

File: app/views/faculties/student_list.html.erb (hunk 2 above)
Comment (britaumich) [reply app/views/faculties/student_list.html.erb]: Oops, I was sure I did that. Thanks for noticing.

**Pull request merged**